vision_datasets/common/data_manifest.py

```diff
@@ -324,6 +342,62 @@ def sample_subset_by_ratio(self, sampling_ratio):
         sampled_images = [self.images[i] for i in sampled_image_ids]
         return DatasetManifest(sampled_images, self.labelmap, self.data_type)
 
+    def sample_n_way_n_shot(self, n_way, n_shot, random_seed=0):
+        """
+        Sample N-way N-shot dataset. Sampled dataset has n_way classes and each classes has n_shot images.
+        Each image will have only 1 class. If the original image has multiple classes, annotations for the other classes will be removed.
+
+        This method can throw a RuntimeError if it failed to generate N-way N-shot dataset.
+
+        Args:
+            n_way (int): The number of classes to sample.
+            n_shot (int): The number of images for each classes.
+
+        Returns:
+            A sampled dataset (DatasetManifest)
+        """
+        if self.is_multitask:
+            raise RuntimeError("N-way N-shot sampling from multi-task is not supported.")
+
+        rng = random.Random(random_seed)
+        labels = [[self._get_cid(c) for c in image.labels] for image in self.images]
+
+        # Get lists of images per class_ids.
+        images_by_cids = collections.defaultdict(list)
+        for i, label in enumerate(labels):
+            for c in set(label):
+                images_by_cids[c].append(i)
+
+        # Get a list of classes that have at least N images, and sample N-way classes.
+        sampled_classes = list(rng.sample([c for c in images_by_cids if len(images_by_cids[c]) >= n_shot], n_way))
```

Comment:
what is no classes or < N_way classes have images >= n_shot?

Reply:
Good point. will add a check